packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.Props.ts

```diff
@@ -52,4 +52,9 @@ export interface ISearchBoxProps extends React.Props<SearchBox> {
    * @defaultvalue labelText
    */
   ariaLabel?: string;
+
+  /**
+   * The boolean which indicates whether to hide search icon.
```

Comment:
Actually, there is 'x' in the TextField. It'll be more effort to add the 'x' than hiding the search icon in SearchBox optionally. Since the control is a SearchBox, we'd like to extend the same of the rest behavior from Fabric.